contrib/win32/win32compat/shell-host.c

```diff
@@ -1181,20 +1172,50 @@ start_withno_pty(int ac, wchar_t **av)
 	return child_exit_code;
 }
 
+int b64_pton(char const *src, u_char *target, size_t targsize);
+
 int 
 wmain(int ac, wchar_t **av)
 {
-	/* create job to hold all child processes */
-	HANDLE job = CreateJobObject(NULL, NULL);
-	JOBOBJECT_EXTENDED_LIMIT_INFORMATION job_info;
-	memset(&job_info, 0, sizeof(JOBOBJECT_EXTENDED_LIMIT_INFORMATION));
-	job_info.BasicLimitInformation.LimitFlags = JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
-	if (!SetInformationJobObject(job, JobObjectExtendedLimitInformation, &job_info, sizeof(job_info)))
+	int pty_requested = 0;
+	wchar_t *cmd = NULL, *cmd_b64 = NULL;
+	{
+		/* create job to hold all child processes */
+		HANDLE job = CreateJobObject(NULL, NULL);
+		JOBOBJECT_EXTENDED_LIMIT_INFORMATION job_info;
+		memset(&job_info, 0, sizeof(JOBOBJECT_EXTENDED_LIMIT_INFORMATION));
+		job_info.BasicLimitInformation.LimitFlags = JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
+		if (!SetInformationJobObject(job, JobObjectExtendedLimitInformation, &job_info, sizeof(job_info)))
+			return -1;
+		CloseHandle(job);
+	}
+
+	if ((ac == 1) || (ac == 2 && wcscmp(av[1], L"-nopty"))) {
+		pty_requested = 1;
+		cmd_b64 = ac == 2? av[1] : NULL;
+	} else if (ac <= 3 && wcscmp(av[1], L"-nopty") == 0)
```

Comment:
can you add some comments.. how the number of arguments are related to "-nopty" arg?